erts/emulator/beam/packet_parser.c

```diff
@@ -533,15 +533,37 @@ http_parse_absoluteURI(PacketHttpURI* uri, const char* uri_ptr, int uri_len)
 
     uri->s1_ptr = uri_ptr;
     uri->port = 0; /* undefined */
-    /* host[:port]  */
     if ((p = memchr(uri_ptr, ':', uri_len)) == NULL) {
         uri->s1_len = uri_len;
     }
+    /* ipv6
+     * eg [::1]:4000
+     * eg [FEDC:BA98:7654:3210:FEDC:BA98:7654:3210]:80
+     */
+    else if (memchr(uri_ptr, '[', uri_len) == uri_ptr) {
+      if ((p = memchr(uri_ptr, ']', uri_len)) != NULL) {
+        int n = (p - uri_ptr) + 1;
+        int port = 0;        
+        uri->s1_len = n;
+        n = uri_len - (n+1);
+        // Need to skip over `]:`
+        p = p + 2;
+        while(n && isdigit((int) *p)) {
```

Comment:
Here we blindly skip over an assumed `:` after `]`.

There are no error cases. Maybe it's better to let port number be `undefined` instead of potentially wrong in case of missing `:`.

Reply:
I added a check here to skip over port parsing if the next character after the `]` isn't a `:`